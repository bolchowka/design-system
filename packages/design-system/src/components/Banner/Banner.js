import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import getMergedClassNames from '../../utils/getMergedClassNames';
import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
import AlertIcon from 'react-material-icon-svg/dist/AlertIcon';
import InformationOutlineIcon from 'react-material-icon-svg/dist/InformationOutlineIcon';
import BlockHelperIcon from 'react-material-icon-svg/dist/BlockHelperIcon';
import CheckCircleIcon from 'react-material-icon-svg/dist/CheckCircleIcon';

import styles from './style.scss';

const cx = classNames.bind(styles);

const baseClass = 'banner';

export const Banner = props => {
  const {
    className,
    closeIcon,
    onClose,
    size,
    text,
    type,
    ...restProps
  } = props;

  const mergedWrapperClassNames = getMergedClassNames(
    cx({
      [`${baseClass}--close`]: true,
      [`${baseClass}--content--${size}`]: size,
      [`${baseClass}--content`]: true,
      [`${baseClass}--text--${size}`]: size,
      [`${baseClass}--wrapper--${type}`]: true,
      [`${baseClass}--wrapper`]: true,
      [`${baseClass}`]: true,
    }),
    className
  );

  const getIconByType = (type) => {
    switch(type) {
      case 'info':
        return <InformationOutlineIcon fill="#4384f5" className={styles['banner-icon']} width="20px" height="20px"/>;
      case 'warning':
        return <AlertIcon fill="#fb9d01" className={styles['banner-icon']} width="20px" height="20px"/>;
      case 'success':
        return <CheckCircleIcon fill="#38c776" className={styles['banner-icon']} width="20px" height="20px"/>;
      case 'error':
        return <BlockHelperIcon fill="#d64646" className={styles['banner-icon']} width="20px" height="20px"/>;
      default:
        return <AlertIcon fill="#fb9d01" className={styles['banner-icon']} width="20px" height="20px"/>;
    }
  }

  return (
    <div className={mergedWrapperClassNames} {...restProps}>
      {
        closeIcon && <CloseIcon onClick={onClose} fill="#424d57" className={styles['banner--icon-close']} data-test="banner-close"/>
      }
      <div className={styles[`banner--content--${size}`]}>
        {getIconByType(type)}
        <p className={styles[`banner--text--${size}`]}>{text}</p>
      </div>      
    </div>
  );
}

Banner.propTypes = {
  closeIcon: PropTypes.bool,
  onClose: PropTypes.func,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  text: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['info', 'warning', 'success', 'error']),
};

Banner.defaultProps = {
  closeIcon: false,
  size: 'small',
  type: 'info',
};

export default Banner;
