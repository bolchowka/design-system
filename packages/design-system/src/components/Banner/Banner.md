### Banner - Notifications

## Usage

### Small

This is appropriate for components from 180px to 400px width. Use it in sidebars, mobile and any other contextual use, where space is the limitation.

Info
```js
<Banner onClose={()=>window.alert('close')}> A description with a <b>maximum of 100</b> characters. That usually means only one or two sentences. </Banner>
```

```js noeditor
<ComponentHtmlMarkup>
<Banner onClose={()=>window.alert('close')}> A description with a maximum of 100 characters. That usually means only one or two sentences. </Banner>
</ComponentHtmlMarkup>
```

Error / blocked
```js
<Banner type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```

```js noeditor
<ComponentHtmlMarkup>
<Banner type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Warning

```js
<Banner type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```

```js noeditor
<ComponentHtmlMarkup>
<Banner type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Success
```js
<Banner type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```

```js noeditor
<ComponentHtmlMarkup>
<Banner type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```

### Medium
This is appropriate for components from 400px to around 800px width.

Max text colum width is 520px

Info
```js
<Banner size="medium" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```

```js noeditor
<ComponentHtmlMarkup>
  <Banner size="medium" onClose={()=>window.alert('close')}>
  A description with a maximum of 100 characters. That usually means only one or two sentences.
  </Banner>
</ComponentHtmlMarkup>
```

Error / blocked
```js
<Banner size="medium" type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="medium" type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Warning
```js
<Banner size="medium" type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="medium" type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Success
```js
<Banner size="medium" type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="medium" type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
### Large

When there is enough space to fit the button on the side, then this component is transforming to Large form.

Max text colum width is 720px

Info
```js
<Banner size="large" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="large" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```

Error / blocked
```js
<Banner size="large" type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="large" type="error" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Warning
```js
<Banner size="large" type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="large" type="warning" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```
Success
```js
<Banner size="large" type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
```
```js noeditor
<ComponentHtmlMarkup>
<Banner size="large" type="success" onClose={()=>window.alert('close')}>A description with a maximum of 100 characters. That usually means only one or two sentences.</Banner>
</ComponentHtmlMarkup>
```

### RWD support

This component is not fully responsive. If you need an RWD support, you can use [Resize Observer](https://web.dev/en/resize-observer/) and apply appropriate styles. Here is an example of a large banner styled as a small one.

```css
.component {
  position: relative;

  .lc-banner--large {
    max-width: 400px;

    .lc-banner__content {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;
      width: 100%;

      p {
        margin: 14px 0 0 0;
        max-width: 246px;
      }
    }
  }
}
```