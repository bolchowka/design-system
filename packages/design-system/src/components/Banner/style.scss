@import '../../scss/variables/colors';

$base-class: 'banner';

.#{$base-class} {
  &--icon__close {
    height: 24px;
    width: 24px;
    cursor: pointer;
    position: absolute;
    top: 17px;
    right: 20px;
  }

  &__content--small {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 100%;
    max-width: 400px;
    
    p {
      margin: 14px 0 0 0;
      max-width: 246px;
    }

  }

  &__content--medium {
    display: flex;
    justify-content: flex-start;
    width: 100%;   
    max-width: 800px;

    p {
      margin: 0 30px 0 16px;
      max-width: 520px;
    }
  }
  
  &__content--large {
    display: flex;
    justify-content: flex-start;
    width: 100%;

    p {
      margin: 0 30px 0 16px;
      max-width: 720px;
    }
  }

  &--wrapper {
    position: relative;
    top: 0;
    left: 0;
    padding: 20px;
    overflow: hidden;
    word-break: break-word ;
  }
  &--wrapper--info {
    box-shadow: inset 0 3px 0 0 #4384f5;
    background-color: #e4f2ff;
  }
  &--wrapper--warning {
    box-shadow: inset 0 3px 0 0 #fb9d01;
    background-color: #fff8e1;
  }
  &--wrapper--success {
    box-shadow: inset 0 3px 0 0 #38c776;
    background-color: #e5f4eb;
  }
  &--wrapper--error {
    box-shadow: inset 0 3px 0 0 #d64646;
    background-color: #fbecef;
  }
}