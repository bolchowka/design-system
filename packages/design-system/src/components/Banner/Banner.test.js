import * as React from 'react';
import { shallow } from 'enzyme';
import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';

import Banner from './Banner';

describe('Banner', () => {
  it('should render small info Banner', () => {
    const component = shallow(<Banner text="Example text" />);

    expect(component).toMatchSnapshot();
  });

  it('should render large Banner', () => {
    const component = shallow(<Banner text="Example text" size="large"/>);

    expect(component).toMatchSnapshot();
  });
  it('should render medium Banner', () => {
    const component = shallow(<Banner text="Example text" size="medium" />);

    expect(component).toMatchSnapshot();
  });
  it('should render warning large Banner', () => {
    const component = shallow(<Banner text="Example text" size="large" type="warning" />);

    expect(component).toMatchSnapshot();
  });
  it('should render info medium Banner', () => {
    const component = shallow(<Banner text="Example text" size="medium" />);

    expect(component).toMatchSnapshot();
  });
  it('should render error small Banner', () => {
    const component = shallow(<Banner text="Example text" type="error" />);

    expect(component).toMatchSnapshot();
  });
  it('should render success large Banner', () => {
    const component = shallow(<Banner text="Example text" type="success" size="large" />);

    expect(component).toMatchSnapshot();
  });
  it('should render Banner with close button', () => {
    const component = shallow(<Banner text="Example text" closeIcon />);

    expect(component).toMatchSnapshot();
  })
  it('should call on close on close button click', () => {
    const onClose = jest.fn();
    const component = shallow(<Banner text="Example text" onClose={onClose} closeIcon />);
    component.find(CloseIcon).simulate('click')
    expect(onClose.mock.calls.length).toEqual(1);
  })
});
