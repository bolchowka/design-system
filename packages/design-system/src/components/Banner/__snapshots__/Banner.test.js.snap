// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render Banner with close button 1`] = `
<div
  className="banner--close banner--content--small banner--content banner--text--small banner--wrapper--info banner--wrapper banner"
>
  <CloseIcon
    className="banner--icon-close"
    data-test="banner-close"
    fill="#424d57"
  />
  <div
    className="banner--content--small"
  >
    <InformationOutlineIcon
      className="banner-icon"
      fill="#4384f5"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--small"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render error small Banner 1`] = `
<div
  className="banner--close banner--content--small banner--content banner--text--small banner--wrapper--error banner--wrapper banner"
>
  <div
    className="banner--content--small"
  >
    <BlockHelperIcon
      className="banner-icon"
      fill="#d64646"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--small"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render info medium Banner 1`] = `
<div
  className="banner--close banner--content--medium banner--content banner--text--medium banner--wrapper--info banner--wrapper banner"
>
  <div
    className="banner--content--medium"
  >
    <InformationOutlineIcon
      className="banner-icon"
      fill="#4384f5"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--medium"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render large Banner 1`] = `
<div
  className="banner--close banner--content--large banner--content banner--text--large banner--wrapper--info banner--wrapper banner"
>
  <div
    className="banner--content--large"
  >
    <InformationOutlineIcon
      className="banner-icon"
      fill="#4384f5"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--large"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render medium Banner 1`] = `
<div
  className="banner--close banner--content--medium banner--content banner--text--medium banner--wrapper--info banner--wrapper banner"
>
  <div
    className="banner--content--medium"
  >
    <InformationOutlineIcon
      className="banner-icon"
      fill="#4384f5"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--medium"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render small info Banner 1`] = `
<div
  className="banner--close banner--content--small banner--content banner--text--small banner--wrapper--info banner--wrapper banner"
>
  <div
    className="banner--content--small"
  >
    <InformationOutlineIcon
      className="banner-icon"
      fill="#4384f5"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--small"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render success large Banner 1`] = `
<div
  className="banner--close banner--content--large banner--content banner--text--large banner--wrapper--success banner--wrapper banner"
>
  <div
    className="banner--content--large"
  >
    <CheckCircleIcon
      className="banner-icon"
      fill="#38c776"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--large"
    >
      Example text
    </p>
  </div>
</div>
`;

exports[`Banner should render warning large Banner 1`] = `
<div
  className="banner--close banner--content--large banner--content banner--text--large banner--wrapper--warning banner--wrapper banner"
>
  <div
    className="banner--content--large"
  >
    <AlertIcon
      className="banner-icon"
      fill="#fb9d01"
      height="20px"
      width="20px"
    />
    <p
      className="banner--text--large"
    >
      Example text
    </p>
  </div>
</div>
`;
